packages/commons/src/record.ts

```diff
@@ -0,0 +1,141 @@
+import { PractitionerRole } from 'fhir/r3'
+import {
+  Bundle,
+  Composition,
+  CorrectionRequestedTask,
+  DocumentReference,
+  Encounter,
+  Location,
+  Observation,
+  Patient,
+  PaymentReconciliation,
+  Practitioner,
+  RelatedPerson,
+  Task,
+  getBusinessStatus,
+  getTaskFromBundle,
+  isCorrectionRequestedTask,
+  isTask,
+  sortTasksDescending
+} from './fhir'
+
+// https://dnlytras.com/blog/nominal-types
+declare const __nominal__type: unique symbol
+declare const __nominal__type__2: unique symbol
+export type Nominal<Type, Identifier extends string> = Type & {
+  readonly [__nominal__type]: Identifier
+}
+
+type RecordBase = Bundle
+export type WaitingForValidationRecord = Nominal<
+  RecordBase,
+  'WaitingForValidation'
+>
+export type ValidatedRecord = Nominal<RecordBase, 'Validated'>
+export type RegisteredRecord = Nominal<
+  Bundle<
+    | Composition
+    | DocumentReference
+    | Encounter
+    | Patient
+    | RelatedPerson
+    | PaymentReconciliation
+    | Task
+    | Practitioner
+    | PractitionerRole
+    | Location
+    | Observation
+  >,
```

Comment:
Definitely! I'd love if it would be possible to have a way of defining that the bundle includes types X, Y, Z at least once. 

Kind of like a tuple but so that items could exist multiple times